kmk/extensions/split.py

```diff
@@ -0,0 +1,111 @@
+import busio
+
+from kmk.extensions import Extension
+from kmk.matrix import intify_coordinate
+
+
+class SplitType:
+    UART = 1
+    I2C = 2  # unused
+    ONEWIRE = 3  # unused
+
+
+class Split(Extension):
```

Comment:
would this benefit from a bit of oop as it seems to overlap in functionality to the `BLE_Split`?

Reply:
Some functions overlap, though some of BLE extends past what I can do with a wire right now. The main reason for not loading anything BLE in this fie is to prevent issues on boards without BLE, and to reduce memory usage, though I'm open to ideas that take these into consideration.

Reply:
Looking back at this further, there are a lot of functions that must act differently, and jamming all of that into a single class would get more difficult to read. When extending this later, I could import functions from the wired split into the wireless where there is overlap, keeping readability, and that allows BLE to stay completely out of memory on wired only boards. Wired keyboards will need to be brought up to spec with BLE, which requires writing a onewire protocol first, and is out of scope for this PR, much the same as doing RGB matrix is out of scope. If I'm wrong, please correct anything here.

Reply:
Makes sense! I appreciate the thought youve put into responding to my comments!